### PR TITLE
[WIP] Adds Language Model and Wikitext2 Dataloader

### PR DESCRIPTION
**Pull request by Panaetius**
Also does small refactorings.

### CODE REVIEW

File: mlbench_core/dataset/imagerecognition/pytorch/dataloader.py [hunk 1]
```diff
@@ -91,25 +90,3 @@ def __init__(self, root, train=True):
             self.root = os.path.join(self.root, 'val')
 
         super().__init__(self.root, transform)
-
-
-def partition_dataset_by_rank(dataset, rank, world_size, distribution='uniform', shuffle=True):
```
Comment (ehoelzl) [inline mlbench_core/dataset/imagerecognition/pytorch/dataloader.py]: This function is used many places in `mlbench-benchmarks` (e.g. in https://github.com/mlbench/mlbench-benchmarks/blob/896d3c4f31aa7205ad9fcedebd5dc46e3b8e36ed/pytorch/imagerecognition/openmpi-cifar10-resnet20-all-reduce/main.py#L78). Maybe this needs to be changed too before merging

Comment (Panaetius) [reply mlbench_core/dataset/imagerecognition/pytorch/dataloader.py]: Yes definitely, at the moment is copied in 2-3 places already which is why i want to move it to a single file. Though unfortunately it doesn't work with torchtext datasets at the moment.

File: mlbench_core/controlflow/pytorch/controlflow.py [hunk 2]
```diff
@@ -37,7 +37,7 @@ def _record_train_batch_stats(batch_idx, loss, output, target, metrics,
 
     # Compute metrics for one batch
     for metric in metrics:
-        metric_value = metric(output, target).item()
+        metric_value = metric(loss, output, target).item()
```
Comment (ehoelzl) [inline mlbench_core/controlflow/pytorch/controlflow.py]: Would this work with the metrics used for [CIFAR10](https://github.com/mlbench/mlbench-benchmarks/blob/896d3c4f31aa7205ad9fcedebd5dc46e3b8e36ed/pytorch/imagerecognition/openmpi-cifar10-resnet20-all-reduce-scaling/main.py#L81)  ? Versions make it okay for now but we need to keep in mind when updating versions

File: mlbench_core/dataset/linearmodels/pytorch/dataloader.py [hunk 3]
```diff
@@ -216,25 +214,3 @@ def load_libsvm_lmdb(name, lmdb_path):
                      target_transform=None, is_image=False)
     return dataset
 
-
-def partition_dataset_by_rank(dataset, rank, world_size, distribution='uniform', shuffle=True):
```
Comment (ehoelzl) [inline mlbench_core/dataset/linearmodels/pytorch/dataloader.py]: Need to change imports in `mlbench-benchmarks`

File: mlbench_core/dataset/imagerecognition/pytorch/dataloader.py (hunk 1 above)
Comment (ehoelzl) [inline mlbench_core/dataset/imagerecognition/pytorch/dataloader.py]: Need to change imports in `mlbench-benchmarks`

File: mlbench_core/dataset/linearmodels/pytorch/dataloader.py (hunk 3 above)
Comment (ehoelzl) [reply mlbench_core/dataset/linearmodels/pytorch/dataloader.py]: @Panaetius Sorry I deleted the comment by mistake.

**Pull request merged**